tests/main.nf.test

```diff
@@ -1,37 +1,122 @@
-nextflow_workflow {
+nextflow_pipeline {
 
     name "Full Integration Tests for MIKROKONDO"
     script "main.nf"
-    workflow "MIKROKONDO"
 
-    test("Should run without failures") {
+    test("Should fail to assembly due to too few reads.") {
 
         when {
             params {
-                input = "samplesheet-staph.csv"
-                platform = "illumina"
+                input = "https://github.com/phac-nml/mikrokondo/raw/dev/tests/data/samplesheets/samplesheet-campy-staph.csv"
                 outdir = "results"
-                mash.mash_sketch = "campy-staph-ecoli.msh"
-                r_contaminants.mega_mm2_idx = "campy.mmi"
+
+                platform = "illumina"
+
+                mash { mash_sketch = "https://github.com/phac-nml/mikrokondo/raw/dev/tests/data/databases/campy-staph-ecoli.msh" }
+                r_contaminants { mega_mm2_idx = "https://github.com/phac-nml/mikrokondo/raw/dev/tests/data/databases/campy.mmi" }
+
+                fastp { args { illumina = "-Q"} }
+                min_reads = 100
+                mash { min_kmer = 1 }
+
                 skip_bakta = true
                 skip_staramr = true
                 skip_mobrecon = true
                 skip_checkm = true
                 skip_raw_read_metrics = true
                 skip_polishing = true
-                max_memory = "4.GB"
-                max_cpus = 2
+
+                max_memory = "2.GB"
+                max_cpus = 1
             }
-            workflow {
-                """
-                // define inputs of the workflow here. Example:
-                // input[0] = file("test-file.txt")
-                """
+        }
+
+        then {
+            assert workflow.success
+            assert path("$launchDir/results").exists()
+
+            // parse output json file
+            def json = path("$launchDir/results/SummaryReport/final_report.json").json
+
+            assert json.CSE.CSE.FastP.summary.sequencing.equals("paired end (150 cycles + 150 cycles)")
+            assert json.CSE.CSE.FastP.summary.before_filtering.total_reads.equals(248)
+            assert json.CSE.CSE.FastP.filtering_result.passed_filter_reads.equals(248)
+            assert json.CSE.CSE.FastP.insert_size.peak.equals(196)
+            assert json.CSE.CSE.FastP.filtering_result.low_quality_reads.equals(0)
+
+            //assert json.CSE.meta.metagenomic.equals(false)  // Currently, this is "null".
+            assert json.CSE.meta.assembly.equals(false)
+            assert json.CSE.meta.hybrid.equals(false)
+            assert json.CSE.meta.single_end.equals(false)
+            assert json.CSE.meta.merge.equals(false)
+            assert json.CSE.meta.downsampled.equals(false)
+            
+            def assembly_path = "$launchDir/results/assembly/length_filtered_contigs/short_filtered.fasta.gz"
```

Comment:
Thanks so much @emarinier .  Could you name this file `CSE_filtered.fasta.gz`? Since `CSE` is the name of the sample in the samplesheet in this case, and so is the prefix for the fasta file.

Reply:
Updated in e16967ef825b161f0df7c752e544452f1ab93525